### PR TITLE
acceptance tests: delete device endpoint tests

### PR DESCRIPTION
**Pull request by kjaskiewiczz**
https://tracker.mender.io/browse/MEN-1286

@maciejmrowiec @mendersoftware/rndity

### CODE REVIEW

File: tests/tests/client.py [hunk 1]
```diff
@@ -90,6 +91,15 @@ def create_user(self, user, auth=None):
 
         return self.client.users.post_users(user=user, _request_options={"headers": auth}).result()
 
+    def delete_user(self, user_id, auth=None, headers={}):
+        if auth is None:
+            auth=self.auth
+
+        headers['Authorization'] = auth['Authorization']
+        # bravado for some reason doesn't issue DELETEs properly (silent failure)
+        rsp = requests.delete(self.make_api_url('/users/{}'.format(user_id)), headers = headers)
```
Comment (bboozzoo) [inline tests/tests/client.py]: `headers=headers`

File: tests/tests/tenantadm.py [hunk 2]
```diff
@@ -40,3 +40,23 @@ def run_fake_create_user(user, status=201):
...
+        if expected_user_id is not None:
+            assert user_id == expected_user_id
+        return (204, {}, '')
+
+    return delete_user
+
+@contextmanager
+def run_fake_delete_user(expected_tenant_id=None, expected_user_id=None):
+    handlers = [
+            ('DELETE', '/api/internal/v1/tenantadm/tenants/(.*)/users/(.*)', fake_delete_user(expected_tenant_id, expected_user_id))
```
Comment (bboozzoo) [inline tests/tests/tenantadm.py]: line break before `fake_delete_user`

File: tests/tests/test_api_management.py [hunk 3]
```diff
@@ -178,3 +178,43 @@ def test_ok(self, tenant_id, api_client_mgmt, init_users_mt):
...
+class TestManagementApiDeleteUserBase:
+    def _do_test_ok(self, api_client_mgmt, init_users, tenant_id=None):
+        auth=None
+        if tenant_id is not None:
+            auth = make_auth("foo", tenant_id)
+
+        rsp = api_client_mgmt.delete_user(init_users[0]['id'], auth)
+        assert rsp.status_code == 204
+        users = api_client_mgmt.get_users(auth)
+        assert len(users) == len(init_users) - 1
```
Comment (bboozzoo) [inline tests/tests/test_api_management.py]: you should also check that user with ID you removed does not exist

File: tests/tests/test_api_management.py [hunk 4]
```diff
@@ -178,3 +178,43 @@ def test_ok(self, tenant_id, api_client_mgmt, init_users_mt):
...
+
+    def _do_test_not_found(self, api_client_mgmt, tenant_id=None):
+        auth=None
+        if tenant_id is not None:
+            auth = make_auth("foo", tenant_id)
+
+        rsp = api_client_mgmt.delete_user('unexisting_id', auth)
```
Comment (bboozzoo) [inline tests/tests/test_api_management.py]: s/unexisting/nonexistent/

File: tests/tests/client.py (hunk 1 above)
Comment (kjaskiewiczz) [inline tests/tests/client.py]: done

File: tests/tests/tenantadm.py (hunk 2 above)
Comment (kjaskiewiczz) [inline tests/tests/tenantadm.py]: done

File: tests/tests/test_api_management.py (hunk 3 above)
Comment (kjaskiewiczz) [inline tests/tests/test_api_management.py]: done

File: tests/tests/test_api_management.py (hunk 4 above)
Comment (kjaskiewiczz) [inline tests/tests/test_api_management.py]: done

**Pull request merged**